framework/test/unit/modules/keys/keys_module.spec.ts

```diff
@@ -944,6 +972,143 @@ describe('keys module', () => {
+		it('should not fail for sender not present in groups', async () => {
+			// Arrange
+			const accounts = cloneDeep(genesisBlock.header.asset.accounts);
+			const mandatoryKeys = [
+				getRandomBytes(32),
+				getRandomBytes(32),
+				getRandomBytes(32),
+			].sort((a, b) => a.compare(b));
+			const optionalKeys = [getRandomBytes(32)];
+
+			accounts[0].keys.numberOfSignatures = 4;
+			accounts[0].keys.mandatoryKeys = mandatoryKeys;
+			accounts[0].keys.optionalKeys = optionalKeys;
+			const gb = (ObjectUtils.mergeDeep({}, genesisBlock, {
+				header: {
+					asset: {
+						accounts,
+					},
+				},
+			}) as unknown) as GenesisBlock<Account<AccountKeys>>;
+
+			// Act & Assert
+			return expect(
+				keysModule.afterGenesisBlockApply({ genesisBlock: gb } as any),
+			).resolves.toBeUndefined();
+		});
+
+		it('should not fail for sender present in groups', async () => {
+			// Arrange
+			const accounts = cloneDeep(genesisBlock.header.asset.accounts);
+			const mandatoryKeys = [
+				Buffer.from('9cabee3d27426676b852ce6b804cb2fdff7cd0b5', 'hex'),
+				getRandomBytes(32),
+				getRandomBytes(32),
+				getRandomBytes(32),
+			].sort((a, b) => a.compare(b));
+			const optionalKeys = [getRandomBytes(32)];
+
+			accounts[61].keys.numberOfSignatures = 4;
+			accounts[61].keys.mandatoryKeys = mandatoryKeys;
+			accounts[61].keys.optionalKeys = optionalKeys;
+			const gb = (ObjectUtils.mergeDeep({}, genesisBlock, {
+				header: {
+					asset: {
+						accounts,
+					},
+				},
+			}) as unknown) as GenesisBlock<Account<AccountKeys>>;
+
+			// Act & Assert
+			return expect(
+				keysModule.afterGenesisBlockApply({ genesisBlock: gb } as any),
+			).resolves.toBeUndefined();
+		});
+
+		it('should not fail for a maximum of 64 keys distributed among "mandatoryKeys" and "optionalKeys"', async () => {
+			// Arrange
+			const accounts = cloneDeep(genesisBlock.header.asset.accounts);
+
+			const mandatoryKeys = [...Array(32).keys()]
+				.map(() => getRandomBytes(32))
+				.sort((a, b) => a.compare(b));
+			const optionalKeys = [...Array(32).keys()]
+				.map(() => getRandomBytes(32))
+				.sort((a, b) => a.compare(b));
+
+			accounts[0].keys.numberOfSignatures = 38;
+			accounts[0].keys.mandatoryKeys = mandatoryKeys;
+			accounts[0].keys.optionalKeys = optionalKeys;
+
+			const gb = (ObjectUtils.mergeDeep({}, genesisBlock, {
+				header: {
+					asset: {
+						accounts,
+					},
+				},
+			}) as unknown) as GenesisBlock<Account<AccountKeys>>;
+
+			// Act & Assert
+			return expect(
+				keysModule.afterGenesisBlockApply({ genesisBlock: gb } as any),
+			).resolves.toBeUndefined();
+		});
+
+		it('should not fail for a maximum of 64 optioonal keys and number of signatures smaller than 64', async () => {
+			// Arrange
+			const accounts = cloneDeep(genesisBlock.header.asset.accounts);
+
+			const optionalKeys = [...Array(64).keys()]
+				.map(() => getRandomBytes(32))
+				.sort((a, b) => a.compare(b));
+
+			accounts[0].keys.numberOfSignatures = 60;
```

Comment:
why 60? they are all optional rite?

Reply:
They are all optional but you need to configure a `numberOfSignatures` >= 1 and I randomly choose 60